components/convert_channel_modal/convert_channel_modal.tsx

```diff
@@ -1,36 +1,30 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-import {trackEvent} from 'actions/telemetry_actions.jsx';
+import {trackEvent} from 'actions/telemetry_actions';
 import Constants from 'utils/constants';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 
-export default class ConvertChannelModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-        * Function called when modal is dismissed
-        */
-        onHide: PropTypes.func.isRequired,
-        channelId: PropTypes.string.isRequired,
-        channelDisplayName: PropTypes.string.isRequired,
-
-        actions: PropTypes.shape({
+type Props ={
```

Comment:
Weird, I expected the linter to complain here, but maybe it's not configured to catch this:
```suggestion
type Props = {
```